langchain/src/types/openai-types.ts

```diff
@@ -59,11 +58,29 @@ export declare interface OpenAIBaseInput {
   openAIApiKey?: string;
 }
 
+// TODO use OpenAI.Core.RequestOptions when SDK is updated to make it available
+export type OpenAICoreRequestOptions<
+  Req extends object = Record<string, unknown>
+> = {
+  path?: string;
+  query?: Req | undefined;
+  body?: Req | undefined;
+  headers?: Record<string, string | null | undefined> | undefined;
+
+  maxRetries?: number;
+  stream?: boolean | undefined;
+  timeout?: number;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  httpAgent?: any;
+  signal?: AbortSignal | undefined | null;
+  idempotencyKey?: string;
+};
+
 export interface OpenAICallOptions extends BaseLanguageModelCallOptions {
   /**
    * Additional options to pass to the underlying axios request.
    */
-  options?: AxiosRequestConfig;
+  options?: OpenAICoreRequestOptions;
```

Comment:
This is going to affect someone out there... Don't think it's worth shimming though, especially since `headers` is thankfully named the same:

https://axios-http.com/docs/req_config